libsolidity/ast/Types.cpp

```diff
@@ -574,13 +628,14 @@ TypePointer IntegerType::binaryOperatorResult(Token::Value _operator, TypePointe
 	if (
 		_other->category() != Category::RationalNumber &&
 		_other->category() != Category::FixedPoint &&
+		_other->category() != Category::Address &&
```

Comment:
Perhaps it would be easier to extract address here and either only check for comparison or call `_other->binaryOperatorResult(_operator, shared_from_this())`? But might also be a little weird....